portfolio/src/main/webapp/miscPage.js

```diff
@@ -0,0 +1,146 @@
+    title: 'Guadalupe River Trail airport terminus',
+  });
+
+  GRTStartMarker.addListener('click', function () {
+    closeInfo();
+    if (window.innerWidth >= 1600) {
+      document.getElementById('map').style.width = '50%';
+    }
+    const mapInfoContainer = document.getElementById('half-content-container');
+    mapInfoContainer.innerHTML =
+      '<a href="/images/IMG_2686.jpg">' +
+      '<img src="/images/IMG_2686.jpg" style="width: 100%" /></a>';
+    mapInfoContainer.appendChild(
+      createPElement('Me at the start of the trail! I ride here frequently.')
+    );
+    mapInfoContainer.appendChild(
+      createPElement(
+        'The bike is a white Poseidon Expressway-SXL, a flat bar ' +
+          'road bike with fixed gears (46T/16T gear ratio), ' +
+          'a 6061 aluminum frame, and 700x25mm tires.'
+      )
+    );
+    mapInfoContainer.appendChild(createCloseButtonElement());
+  });
+
+  GRTBridgeMarker.addListener('click', function () {
+    closeInfo();
+    if (window.innerWidth >= 1600) {
+      document.getElementById('map').style.width = '50%';
+    }
+    const mapInfoContainer = document.getElementById('half-content-container');
+    mapInfoContainer.innerHTML =
+      '<a href="/images/IMG_2744.jpg">' +
+      '<img src="/images/IMG_2744.jpg" style="width: 100%" /></a>';
+    mapInfoContainer.appendChild(
+      createPElement(
+        'A bridge across the Guadalupe River connecting ' +
+          'the lower and upper trails.'
+      )
+    );
+    mapInfoContainer.appendChild(createCloseButtonElement());
+  });
+
+  GRTAirportMarker.addListener('click', function () {
+    closeInfo();
+    if (window.innerWidth >= 1600) {
+      document.getElementById('map').style.width = '50%';
+    }
+    const mapInfoContainer = document.getElementById('half-content-container');
+    mapInfoContainer.innerHTML =
+      '<a href="/images/IMG_2751.jpg">' +
+      '<img src="/images/IMG_2751.jpg" style="width: 100%" /></a>';
+    mapInfoContainer.appendChild(
+      createPElement(
+        'The trail passes right next to Mineta San Jose ' +
+          'International Airport!'
+      )
+    );
+    mapInfoContainer.appendChild(createCloseButtonElement());
+  });
+
+  const GRTPath = new google.maps.Polyline({
+    path: GRTCoordinates,
+    geodesic: true,
+    strokeColor: '#FF0000',
+    strokeOpacity: 1.0,
+    strokeWeight: 2,
+  });
+
+  GRTPath.setMap(map);
+}
+
+function closeInfo() {
+  document.getElementById('half-content-container').innerHTML = '';
+  document.getElementById('map').style.width = '100%';
+}
+
+// Holds the current indices of both slideshows
+let curShowIdxs = [0, 0];
+
+/**
+ * Change picture for either slideshow.
+ */
+function changeSlide(slideshowNum, direction) {
+  const show = [
+    ['IMG_8732.jpg', 'IMG_8795.jpg', 'IMG_8819.jpg'],
+    ['IMG_2715.jpg', 'IMG_2718.jpg', 'IMG_2722.jpg'],
+  ];
+  curShowIdxs[slideshowNum] =
+    (((curShowIdxs[slideshowNum] + direction) % 3) + 3) % 3;
```

Comment:
Isn't ((x % 3) + 3) % 3 equivalent to just x % 3?

Reply:
For -1 % 3, JS returns -1 instead of 2.

Reply:
Fun fact: different languages will return different values for modulo of negative numbers....

If there isn't a way to simplify this then I would suggest adding a comment on what this is trying to do because it is definitely hard to parse.  If someone had to modify this then they would definitely probably get it wrong.

You could also consider breaking it apart as:

curShowIdxs[slideshowNum] = (curShowIdxs[slideshowNum] + direction) % 3;
// We need to handle negative numbers as -1 % 3 is -1.
if (curShowIdxs[slideshowNum] < 0) {
  curShowIdxs[slideshowNum] += 3;
}

Reply:
Yeah, thanks for the explanation. JavaScript is a language full of hidden delights like this. 😑

Looking online, it seems like your approach is idiomatic in JavaScript. As Mike said, a comment would help, or you could put that math in your own mod() function with a comment there.